Common/src/CConfig.cpp

```diff
@@ -4652,6 +4658,53 @@ void CConfig::SetPostprocessing(unsigned short val_software, unsigned short val_
     }
   }
 
+  /*--- Check that if the wall roughness array are compatible and set deafult values if needed. ---*/
+
+  if ((nMarker_HeatFlux > 0) || (nMarker_Isothermal > 0)) {
+    /*--- If the config option is not declared, then assume smooth walls and set roughness to zero. ---*/
+    if (nWall_Types != nMarker_HeatFlux + nMarker_Isothermal) {
+      /*--- If this option is not used at all, assume all walls are smooth and set roughness height to zero. ---*/
+      if (nWall_Types == 0) {
+        nWall_Types = nMarker_HeatFlux + nMarker_Isothermal;
+        if (Roughness_Height != NULL) Roughness_Height = NULL;
+        Roughness_Height = new su2double [nWall_Types];
+        if (Kind_Wall != NULL) Kind_Wall = NULL;
+        Kind_Wall = new unsigned short [nWall_Types];
```

Comment:
This raises an eyebrow, if the thing is not null it was either not initialized properly or it was already allocated in which case it becomes a memory leak.

Reply:
Thanks for the catch. Turns out I hadn't initialized one of the pointers (Kind_Wall) to NULL. The other one is initialized earlier. I put the if condition mostly as an insurance. Valgrind does not report any errors when I remove the if condition. Should I just remove it?

Reply:
Yes please. That method is called from the constructors so there is no risk of leaking (at the moment...). If the purpose of the check is to avoid a memory leak you need to do `delete [] Roughness_Height;`
The null-check is actually not required because a) operator delete does it; b) if the pointer is not null because it was not initialized you get an even worse bug (deleting someone else's memory).

In summary, use std::vector :)

Reasons not to use std::vector: it is usually implemented as 3 pointers (begin end and capacity) so if you have many small arrays that share a common size (the case of CConfig) it is a bit more efficient (albeit error prone) to use c-style arrays.